cabal.project.freeze

```diff
@@ -0,0 +1,290 @@
+active-repositories: hackage.haskell.org:merge
+constraints: any.Cabal ==3.6.3.0,
+             any.Glob ==0.10.2,
+             any.OneTuple ==0.3.1,
+             any.Only ==0.1,
+             any.QuickCheck ==2.14.2,
+             QuickCheck -old-random +templatehaskell,
+             any.StateVar ==1.2.2,
+             any.adjunctions ==4.4.2,
+             any.aeson ==2.0.3.0,
+             aeson -cffi +ordered-keymap,
+             any.aeson-pretty ==0.8.9,
+             aeson-pretty -lib-only,
+             any.alex ==3.2.7.4,
```

Comment:
This was manually down-graded to 3.2.7.4 to fix the `format` derivation build. The reason is two-fold:
1. The constraints from `cabal.project.freeze` are being picked up when building `fourmolu` 
2. constraints on tool versions (`--constraint="alex == 3.2.7.4" in this case`)  are ignored by `cabal freeze`

These is currently manual work (the script will try to bump `alex` to the latest available version). Hopefully next time we upgrade the resolver we won't have trouble with it. (just have to make sure `hackage.nix` index state is not too old for our choice of resolver).